code/modules/mining/lavaland/loot/hierophant_loot.dm

```diff
@@ -114,6 +114,10 @@
 	if(user.is_in_active_hand(src) && user.is_in_inactive_hand(src)) //you need to hold the staff to teleport
 		to_chat(user, "<span class='warning'>You need to hold the club in your hands to [beacon ? "teleport with it":"detach the beacon"]!</span>")
 		return
+	var/area/A = get_area(user)
+	if(A.tele_proof)
+		to_chat(user, "<span class='warning'>[src] sparks and fizzles.</span>")
+		return
```

Comment:
This should probably be made into a helper instead of using basically the same thing everywhere.

Reply:
How about the lavaland items just using do_teleport like most teleportation items?
do_teleport already respects tele_proof.
I could make a helper for the cases that don't use do_teleport... but if you want standardization, then just making the lavaland items use do_teleport would probably be better.

Reply:
These aren't really designed to work with those--not to mention it'll break things if you're using a bag of holding.

Reply:
Fair point. I had not considered miners using bluespace bags. That could get wonky.

Updated to use a helper instead.